### ISSUE TITLE
Custom options for PageRender

### ISSUE THREAD

**Issue by yckart** [ryancramerdesign/ProcessWire#1680]
Would nice to be able to pass some kind of custom data into the new `render`-methods. Like an array as already possible with `wireRenderFile`/`wireIncludeFile`. What do you think?

For example:

```php
$page->render('title', [
  'what' => 'ever'
]);
```

**Comment by yckart** [comment]
Whoops, missed that https://github.com/ryancramerdesign/ProcessWire/blob/356a6a28a98de516c7d802156972b6f5a47f059f/wire/modules/PageRender.module#L345-L358 I should look in the core before I ask!

**Comment by Toutouwai** [comment]
Related forum threads:
https://processwire.com/talk/topic/12450-field-rendervalue/
https://processwire.com/talk/topic/13958-help-understanding-render-and-rendervalue/

**Comment by BernhardBaumrock** [comment]
should already be possible like this:
`$page->renderValue([
  'page' => $page,
  'what' => 'ever'
]);`
and in the template file:
`$page = $value['page'];
$what = $value['what'];`

**Comment by Toutouwai** [comment]
Yeah, but this feature request and my duplicate referenced above are about $page->render('some_field') / $page->renderField(), not $page->renderValue().

**Comment by BernhardBaumrock** [comment]
but i agree that this feels a little bit uncomfortable! i would also prefer to have the variables available with their names of the associative array if possible:

```
$page->renderValue([
  'what' => 'ever',
  'and' => 'so on',
], 'your-view');
```
and then having it available like this:
```
echo $what; // ever
echo $and; // so on
```

**Comment by Toutouwai** [comment]
You can use [extract()](http://php.net/manual/en/function.extract.php) on $value to get $what, $and, etc.

**Comment by BernhardBaumrock** [comment]
Thank you didn't know about that :)
